Utils/trigger_private_build.py

```diff
@@ -0,0 +1,125 @@
+import re
+import os
+import sys
+import json
+import time
+import argparse
+import requests
+import logging
+import demisto_sdk.commands.common.tools as tools
+
+# disable insecure warnings
+requests.packages.urllib3.disable_warnings()
+
+INFRASTRUCTURE_FILES = ['Tests/scripts/validate_premium_packs.sh', 'Tests/scripts/validate_premium_packs.py',
+                        'Tests/scripts/validate_index.py']
+INFRASTRUCTURE_FOLDERS = ['Tests/private_build', 'Tests/Marketplace']
+
+TRIGGER_BUILD_URL = 'https://api.github.com/repos/demisto/content-private/dispatches'
+GET_DISPATCH_WORKFLOWS_URL = 'https://api.github.com/repos/demisto/content-private/actions/runs'
+WORKFLOW_HTML_URL = 'https://github.com/demisto/content-private/actions/runs'
+
+PRIVATE_REPO_WORKFLOW_ID_FILE = 'PRIVATE_REPO_WORKFLOW_ID.txt'
+
+
+def get_modified_files(branch_name):
+    files = []
+    files_string = tools.run_command("git diff --name-status origin/master...{0}".format(branch_name))
+    for line in files_string.split("\n"):
+        if line:
+            _, file_path = line.split(maxsplit=1)
+            if file_path:
+                files.append(file_path)
+    return files
+
+
+def is_infrastructure_change(modified_files):
+    for file in modified_files:
+        if file in INFRASTRUCTURE_FILES:
+            return True
+
+        path = os.path.dirname(file)
+        for dir_path in INFRASTRUCTURE_FOLDERS:
+            if path.startswith(dir_path):
+                return True
+    return False
+
+
+def get_dispatch_workflows_ids(bearer_token, branch):
+    res = requests.request("GET",
+                           GET_DISPATCH_WORKFLOWS_URL,
+                           headers={'Authorization': bearer_token},
+                           params={'branch': branch, 'event': 'repository_dispatch'},
+                           verify=False)
+    if res.status_code != 200:
+        logging.error(f'Failed to gets private repo workflows, request to '
+                      f'{GET_DISPATCH_WORKFLOWS_URL} failed with error: {str(res.content)}')
+        sys.exit(1)
+
+    try:
+        workflows = json.loads(res.content)
+    except ValueError:
+        logging.error('Enable to parse private repo workflows response')
+        sys.exit(1)
+
+    workflows = workflows.get('workflow_runs', [])
+    return [workflow.get('id') for workflow in workflows]
+
+
+def main():
+    # get github_token parameter
+    arg_parser = argparse.ArgumentParser()
+    arg_parser.add_argument('--github-token', help='Github token')
+    args = arg_parser.parse_args()
+    bearer_token = 'Bearer ' + args.github_token
+
+    # get branch name
+    branches = tools.run_command("git branch")
+    branch_name_reg = re.search(r"\* (.*)", branches)
+    branch_name = branch_name_reg.group(1)
```

Comment:
you can use `CIRCLE_BRANCH` instead
@hod-alpert what do you think is better?

Reply:
I think `CIRCLE_BRANCH` should do it better.